api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -426,6 +426,7 @@ type HostedClusterSpec struct {
 	//
 	// If a given service is not present in this list, it will be exposed publicly
 	// by default.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Services is immutable. Changes might result in unpredictable and disruptive behavior."
```

Comment:
Should we go ahead and explicitly state whether this is required or not? ([HOSTEDCP-1825](https://issues.redhat.com/browse/HOSTEDCP-1825))

Reply:
every field is required unless specified otherwise. If we want to signal that explicitly consistently (which I think is a good thing), let's put a dedicated separate PR for it.